x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearning.java

```diff
@@ -2263,4 +2274,63 @@ public void signalShutdown(Collection<String> shutdownNodeIds) {
             mlLifeCycleService.get().signalGracefulShutdown(shutdownNodeIds);
         }
     }
+
+    @Override
+    public Map<String, Mapper.TypeParser> getMappers() {
+        return Map.of(SemanticTextFieldMapper.CONTENT_TYPE, SemanticTextFieldMapper.PARSER);
+    }
+
+    @Override
+    public Optional<Pipeline> getIngestPipeline(IndexMetadata indexMetadata, Processor.Parameters parameters) {
```

Comment:
Add a new getIngestPipeline() method to IngestPlugin. This method allows plugins to specify a pipeline to be used for ingestion on an index.

ML Plugin creates an inference pipeline when there are semantic_text field types.